src/perl/lib/wtsi_clarity/epp/generic/worksheet_attacher.pm

```diff
@@ -60,7 +60,7 @@ override 'run' => sub {
   my $stamp = _get_stamp($containers_data, $self->request->user);
 
   # pdf generation
-  my $pdf_data = $self->_get_pdf_data($containers_data, $stamp, $type_data);
+  my $pdf_data = _get_pdf_data($containers_data, $stamp, $type_data);
```

Comment:
I guess, this would be a quick fix, and it should work, but with this change we would go backwards regarding to code quality.
From http://perldoc.perl.org/perlobj.html:
"Perl does not provide any special syntax for defining a method. A method is simply a regular subroutine, and is declared with sub. What makes a method special is that it expects to receive either an object or a class name as its first argument."
I also understand - as I reviewed this class -, that this class needs a lot of fix like this.
I would do those changes, too. Sorry. :(